src/TorchSharp/Utils/tensorboard/SummaryWriter.cs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Collections.Generic;
+using System.Threading;
+using Google.Protobuf;
+using Tensorboard;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        public static partial class utils
+        {
+            public static partial class tensorboard
+            {
+                /// <summary>
+                /// Writes entries directly to event files in the log_dir to be consumed by TensorBoard.
+                /// 
+                /// The SummaryWriter class provides a high-level API to create an event file in a given directory and add summaries and events to it.The class updates the file contents asynchronously.
+                ///
+                /// This allows a training program to call methods to add data to the file directly from the training loop, without slowing down training.
+                /// </summary>
+                /// <param name="log_dir">
+                /// Save directory location. Default is runs/CURRENT_DATETIME_HOSTNAME, which changes after each run.
+                /// Use hierarchical folder structure to compare between runs easily. e.g. pass in �runs/exp1�, �runs/exp2�, etc.
+                /// for each new experiment to compare across them
+                /// </param>
+                /// <param name="filename_suffix">Suffix added to all event filenames in the log_dir directory.</param>
+                /// <param name="createRunName">Create a time-based run name, even if log_dir is specified.</param>
+                public static Modules.SummaryWriter SummaryWriter(string log_dir = null, string filename_suffix = null, bool createRunName = false)
+                {
+                    if (createRunName && !String.IsNullOrEmpty(log_dir)) {
+                        log_dir = CreateRunName(log_dir);
+                    }
+
+                    return new Modules.SummaryWriter(log_dir, filename_suffix);
+                }
+
+                internal static string CreateRunName(string log_dir)
+                {
+                    var now = DateTime.Now;
+                    var name = $"{Modules.SummaryWriter._months[now.Month - 1]}{now.Day}_{now.Hour}-{now.Minute}-{now.Second}_{System.Net.Dns.GetHostName()}";
```

Comment:
>_months [](http://example.com/codeflow?start=56&length=7)

nit: may be can get rid of `_months` and use `CultureInfo.InvariantCulture.DateTimeFormat.AbbreviatedMonthNames` instead?

Reply:
Thanks for that hint -- I wasn't aware of its existence.